test-utils/testlib/src/user/runtime_user.rs

```diff
@@ -109,38 +109,50 @@ impl RuntimeUser {
         }
     }
 
-    fn get_recursive_transaction_results(&self, hash: &CryptoHash) -> Vec<TransactionLogView> {
-        let result = self.get_transaction_result(hash);
-        let receipt_ids = result.receipts.clone();
-        let mut transactions = vec![TransactionLogView { hash: hash.clone().into(), result }];
+    fn get_recursive_transaction_results(
+        &self,
+        hash: &CryptoHash,
+    ) -> Vec<ExecutionOutcomeWithIdView> {
+        let outcome = self.get_transaction_result(hash);
+        let receipt_ids = outcome.receipt_ids.clone();
+        let mut transactions = vec![ExecutionOutcomeWithIdView { id: (*hash).into(), outcome }];
         for hash in &receipt_ids {
             transactions
                 .extend(self.get_recursive_transaction_results(&hash.clone().into()).into_iter());
         }
         transactions
     }
 
-    fn get_final_transaction_result(&self, hash: &CryptoHash) -> FinalTransactionResult {
-        let transactions = self.get_recursive_transaction_results(hash);
-        let status = if transactions
-            .iter()
-            .find(|t| &t.result.status == &TransactionStatus::Failed)
-            .is_some()
-        {
-            FinalTransactionStatus::Failed
-        } else if transactions
-            .iter()
-            .find(|t| &t.result.status == &TransactionStatus::Unknown)
-            .is_some()
+    fn get_final_transaction_result(&self, hash: &CryptoHash) -> FinalExecutionOutcomeView {
```

Comment:
isn't this duplicate code?

Reply:
Yeah, I copied it here. The only difference this one is not a `Result` type